File: code_challenge_solution/java/199 Text to Hexadecimal or Binary.java
```java
import java.util.Scanner;

public class TextConverter {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the text to convert: ");
        String text = scanner.nextLine();
        
        System.out.println("Text to Hexadecimal: ");
        String hex = convertToHexadecimal(text);
        System.out.println(hex);
        
        System.out.println("Text to Binary: ");
        String binary = convertToBinary(text);
        System.out.println(binary);
    }
    
    public static String convertToHexadecimal(String text) {
        StringBuilder hexBuilder = new StringBuilder();
        
        for (char character : text.toCharArray()) {
            hexBuilder.append(Integer.toHexString((int) character));
            hexBuilder.append(" ");
        }
        
        return hexBuilder.toString();
    }
    
    public static String convertToBinary(String text) {
        StringBuilder binaryBuilder = new StringBuilder();
        
        for (char character : text.toCharArray()) {
            binaryBuilder.append(String.format("%8s", Integer.toBinaryString((int) character)).replace(' ', '0'));
            binaryBuilder.append(" ");
        }
        
        return binaryBuilder.toString();
    }
}
```

File: code_challenge_solution/python/199 Text to Hexadecimal or Binary.py
text = input("Enter the text: ")
convert_type = input("Enter 'h' for hexadecimal or 'b' for binary conversion: ")

if convert_type == 'h':
    hex_value = ''.join(hex(ord(x))[2:] for x in text)
    print("Hexadecimal: " + hex_value)
elif convert_type == 'b':
    binary_value = ''.join(format(ord(x), '08b') for x in text)
    print("Binary: " + binary_value)
else:
    print("Invalid conversion type. Please enter 'h' or 'b'.")